parlai/mturk/webapp/server.py

```diff
@@ -0,0 +1,375 @@
+# Copyright 2017-present, Facebook, Inc.
+# All rights reserved.
+#
+# This source code is licensed under the license found in the
+# LICENSE file in the root directory of this source tree.
+
+"""ParlAI Server file"""
+
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+from __future__ import unicode_literals
+
+import argparse
+import inspect
+import json
+import logging
+import os
+import time
+import traceback
+
+import tornado.ioloop
+import tornado.web
+import tornado.websocket
+import tornado.escape
+
+
+from parlai.mturk.core.mturk_data_handler import MTurkDataHandler
+
+DEFAULT_PORT = 8095
+DEFAULT_HOSTNAME = "localhost"
+DEFAULT_DB_FILE = 'pmt_data.db'
+IS_DEBUG = True
+
+here = os.path.abspath(os.path.dirname(__file__))
+
+
+def row_to_dict(row):
+    return (dict(zip(row.keys(), row)))
+
+
+def get_rand_id():
+    return str(hex(int(time.time() * 10000000))[2:])
+
+
+def ensure_dir_exists(path):
+    """Make sure the parent dir exists for path so we can write a file."""
+    try:
+        os.makedirs(os.path.dirname(path))
```

Comment:
are you going for `os.makedirs(os.path.dirname(path), exists_ok=True)`?

Reply:
Oh that works a lot better than this. Thanks for the note!